pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2364,19 +2364,19 @@ public CompletableFuture<Boolean> isTopicNsOwnedByBrokerAsync(TopicName topicNam
 
     public CompletableFuture<Void> checkTopicNsOwnership(final String topic) {
         TopicName topicName = TopicName.get(topic);
+        final var namespaceService = pulsar.getNamespaceService();
 
-        return pulsar.getNamespaceService().checkTopicOwnership(topicName)
-                .thenCompose(ownedByThisInstance -> {
+        return namespaceService.getBundleAsync(topicName).thenCompose(bundle ->
+                namespaceService.checkBundleOwnership(topicName, bundle).thenCompose(ownedByThisInstance -> {
                     if (ownedByThisInstance) {
                         return CompletableFuture.completedFuture(null);
                     } else {
-                        String msg = String.format("Namespace bundle for topic (%s) not served by this instance:%s. "
-                                        + "Please redo the lookup. Request is denied: namespace=%s",
```

Comment:
ServiceUnitStateChannelTest is failing since this was removed. 
It might be good to preserve some advice in the warning message. I'm not saying that the previous one is optimal, but it could be useful to have something in the message that it's a transient error and that a new lookup should fix the problem.